DepositApi/Controllers/CalculateDepositApiController.cs

```diff
@@ -13,35 +13,37 @@ namespace DepositApi.Controllers
 {
     [ApiController]
     [Route("/api/calculate")]
-    public class DepositCalcApiController : ControllerBase
+    public class DepositCalculationApiController : ControllerBase
     {
         private readonly IDepositService depositService;
         private readonly IMapper mapper;
 
-        public DepositCalcApiController(IDepositService depositService, IMapper mapper)
+        public DepositCalculationApiController(IDepositService depositService, IMapper mapper)
         {
             this.depositService = depositService;
             this.mapper = mapper;
         }
 
         [HttpGet]
-        public ActionResult Get(DepositModel deposit)
+        public async Task<ActionResult> Get(DepositModel deposit)
         {
             var validator = new DepositValidator();
```

Comment:
Please, do not create validator instance directly. Use DI or better implement validation action filter. With validation action filter you will not have to perform any validation staff and the code will be cleaner.
[AddFluentValidation](https://docs.fluentvalidation.net/en/latest/aspnet.html#getting-started)
[Validation Action Filter example](https://berilkavakli.medium.com/fluent-validation-on-net-core-api-3-1-f01ff4a4c6f5)